### ISSUE TITLE
Possibly confusing error message for E0251/E0252 when importing a tuple struct

### ISSUE THREAD

**Issue by nham** [rust-lang/rust#25392]
This code:

```rust
use foo::Bar;
use bar::Bar;

fn main() { }

mod foo {
    pub struct Bar(u16);
}

mod bar {
    pub struct Bar(u8);
}
```

correctly fails to compile, ~~but it gives an extra wrong error message about "a value named `Bar`"~~ (NOTE: this message is actually correct. It's the constructor of the type. See below):

```
E0252_extra.rs:2:5: 2:13 error: a value named `Bar` has already been imported in this module [E0252]
E0252_extra.rs:2 use bar::Bar;
                     ^~~~~~~~
E0252_extra.rs:2:5: 2:13 error: a type named `Bar` has already been imported in this module [E0252]
E0252_extra.rs:2 use bar::Bar;
                     ^~~~~~~~
error: aborting due to 2 previous errors
```

Note that for non-tuple structs, the message is:

```rust
use foo::Baz;
use bar::Baz;

fn main() { }

mod foo {
    pub struct Baz { x: bool }
}

mod bar {
    pub struct Baz { x: u8 }
}
```

```
E0252_extra.rs:2:5: 2:13 error: a type named `Baz` has already been imported in this module [E0252]
E0252_extra.rs:2 use bar::Baz;
                     ^~~~~~~~
error: aborting due to previous error
```

**Comment by sfackler** [comment]
The message is perhaps confusing but it's correct. `Bar` is both a type and a value (the value is the constructor function).

**Comment by nham** [comment]
Ohhhh, I see. Thanks! I will leave this open for now in case someone can think of a way to improve it.

**Comment by nham** [comment]
Closing this issue. The error message below (as of 1.8) seems perfectly fine and I do not see how it can be improved upon.

```
<anon>:2:5: 2:13 error: a value named `Bar` has already been imported in this module [E0252]
<anon>:2 use bar::Bar;
             ^~~~~~~~
<anon>:2:5: 2:13 help: see the detailed explanation for E0252
<anon>:1:5: 1:13 note: previous import of `Bar` here
<anon>:1 use foo::Bar;
             ^~~~~~~~
<anon>:2:5: 2:13 error: a type named `Bar` has already been imported in this module [E0252]
<anon>:2 use bar::Bar;
             ^~~~~~~~
<anon>:2:5: 2:13 help: see the detailed explanation for E0252
<anon>:1:5: 1:13 note: previous import of `Bar` here
<anon>:1 use foo::Bar;
             ^~~~~~~~
error: aborting due to 2 previous errors
```
